NightscoutServiceKit/RemoteCommands/V1/Notifications/BolusRemoteNotification.swift

```diff
@@ -10,35 +10,34 @@ import Foundation
 import LoopKit
 
 public struct BolusRemoteNotification: RemoteNotification, Codable {
-    
+
     public let amount: Double
     public let remoteAddress: String
     public let expiration: Date?
     public let sentAt: Date?
-    public let otp: String
+    public let otp: String?
+    public let enteredBy: String?
```

Comment:
The answer to "who sent this command?" should ideally be determined by knowing which OTP code was matched, and showing the Loop user a name that the Loop user entered for that OTP pairing when the authorization was initially created.

So when Riley shares her Loop data with me, she puts in "Dad", and then the qrcode is shown and associated with "Dad". I pair that in LCG and when a remote bolus is received by Loop, if it matches my OTP, then Riley gets a notification that "Dad" sent a bolus command.

Reply:
Otherwise it's possible for caregivers to spoof.

Reply:
Yeah that makes sense. This is currently a field in NS but we could eliminate field someday with the support you mentioned. I'm actually not using this field here at the moment but just noticed it was part of the payload.